src/auxiliaryblockrequest.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "auxiliaryblockrequest.h"
+
+#include "chainparams.h"
+#include "validation.h"
+#include "validationinterface.h"
+#include "ui_interface.h"
+
+#include <exception>
+
+static const unsigned int MAX_BLOCK_TO_PROCESS_PER_ITERATION = 5;
+
+std::shared_ptr<CAuxiliaryBlockRequest> currentBlockRequest; //thread-safe pointer (CAuxiliaryBlockRequest, the object, is also lock-free)
+
+CAuxiliaryBlockRequest::CAuxiliaryBlockRequest(std::vector<const CBlockIndex*> vBlocksToDownloadIn, int64_t createdIn, bool passThroughSignalsIn, const std::function<bool(std::shared_ptr<CAuxiliaryBlockRequest>, const CBlockIndex *pindex)> progressCallbackIn) : vBlocksToDownload(vBlocksToDownloadIn), created(createdIn), passThroughSignals(passThroughSignalsIn), progressCallback(progressCallbackIn)
+{
+    fCancelled = false;
+    requestedUpToSize = 0;
+    processedUpToSize = 0;
+
+    NotifyUI();
+}
+
+void CAuxiliaryBlockRequest::NotifyUI()
+{
+    // Notify UI
+    uiInterface.NotifyAuxiliaryBlockRequestProgress(this->created, this->vBlocksToDownload.size(), this->amountOfBlocksLoaded(), this->processedUpToSize);
+}
+
+void CAuxiliaryBlockRequest::processWithPossibleBlock(const std::shared_ptr<const CBlock> pblock, CBlockIndex *pindex)
+{
+    // don't process anything if the request was cancled
```

Comment:
Typo found by `codespell`: cancled

Reply:
This should be fixed throughout in this PR :-)

Reply:
Maybe we need a linter?

Reply:
@Sjors Please review #13954 which add a `codespell` linter :-)